docs/avro.rst

```diff
@@ -145,22 +98,94 @@ simply dropped.  Had the new schema dropped the original field ``favorite_number
 forward compatible with the original user schema since we wouldn't know how to fill in the value for ``favorite_number``
 for the new data because the original schema did not specify a default value for that field.
 
+If the compatibility is set to ``FORWARD_TRANSITIVE`` (not just ``FORWARD``), then it means that consumers using all previousely registered schemas (not just the latest schema) can read data written by producers using the new schema.
+
 
 .. _avro-full_compatibility:
 
 Full Compatibility
 ^^^^^^^^^^^^^^^^^^
 
-Full compatibility means schemas are backward **and** forward compatible.
+``FULL`` compatibility means schemas are backward **and** forward compatible.
+This is when the change to the schema is both forward compatible and backward compatible. In some data formats, such as JSON, there are no full-compatible changes. Every modification is either only forward or only backward compatible. But in other data formats, like Avro, you can define fields with default values. In that case adding or removing a field with a default value is a fully compatible change.
 
 To support both previous use cases on the same data, we can evolve the schemas in a fully compatible way: old data can
 be read with the new schema, and new data can also be read with the old schema.
 
+If the compatibility is set to ``FULL_TRANSITIVE`` (not just ``FULL``), then it means that the new schema is forward and backward compatible with all previously registered schemas (not just the latest schema).
```

Comment:
How about this: FULL_TRANSITIVE compatible schema means that both of these are true:
1. All versions (X, X-1, ..., 1) of the schema can be used to decode data encoded with version X of the schema.
2. Version X of the schema can be used to decode data encoded with all versions (X, X-1, ..., 1) of the schema.

In other word, FULL_TRANSITIVE compatibility (not just ``FULL``) means that the new schema is forward and backward compatible with all previously registered schemas (not just the latest schema).

Reply:
Good idea. I've adapted this suggestion